components/tidb_query/src/codec/mysql/time/mod.rs

```diff
@@ -898,38 +898,43 @@ impl<T: std::io::Write> TimeEncoder for T {}
 pub trait TimeEncoder: NumberEncoder {
     fn encode_time(&mut self, v: &Time) -> Result<()> {
         if !v.is_zero() {
-            self.encode_u16(v.time.year() as u16)?;
+            self.encode_u32_le(v.time.hour() as u32)?;
+            self.encode_u32_le(v.time.nanosecond() / 1000)?;
+            self.encode_u16_le(v.time.year() as u16)?;
             self.write_u8(v.time.month() as u8)?;
             self.write_u8(v.time.day() as u8)?;
-            self.write_u8(v.time.hour() as u8)?;
             self.write_u8(v.time.minute() as u8)?;
             self.write_u8(v.time.second() as u8)?;
-            self.encode_u32(v.time.nanosecond() / 1000)?;
         } else {
-            let len = mem::size_of::<u16>() + mem::size_of::<u32>() + 5;
+            let len = mem::size_of::<u16>() + 2 * mem::size_of::<u32>() + 4;
             let buf = vec![0; len];
             self.write_all(&buf)?;
         }
+        self.encode_u16_le(0 as u16)?;
+        // above 16 bytes
 
         let tp: FieldTypeTp = v.time_type.into();
         self.write_u8(tp.to_u8().unwrap())?;
-        self.write_u8(v.fsp).map_err(From::from)
+        self.write_u8(v.fsp)?;
+        self.encode_u16_le(0 as u16).map_err(From::from)
+        // above 20 bytes
     }
 }
 
 pub trait TimeDecoder: NumberDecoder {
-    /// `decode_time` decodes time encoded by `encode_time` for Chunk format.
+    /// `decode` decodes time encoded by `encode_time` for Chunk format.
     fn decode_time(&mut self) -> Result<Time> {
-        let year = i32::from(self.read_u16()?);
-        let buf = self.read_bytes(5)?;
-        let (month, day, hour, minute, second) = (
+        let hour = self.read_u32_le()?;
```

Comment:
Could you provide a reference of TiDB's implementation so that we can cross verify this logic?